src/main/java/domain/LottoTickets.java

```diff
@@ -0,0 +1,59 @@
+package domain;
+
+import domain.strategy.NumberGenerateStrategy;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class LottoTickets {
+    private final List<LottoTicket> lottoTickets;
+    private final int selfPurchaseCount;
+
+    private LottoTickets(List<LottoTicket> lottoTickets, int selfPurchaseCount) {
+        this.lottoTickets = lottoTickets;
+        this.selfPurchaseCount = selfPurchaseCount;
+    }
+
+    public static LottoTickets generateAutoTickets(LottoMoney autoPurchaseLottoMoney,
+                                                   NumberGenerateStrategy numberGenerateStrategy) {
+        int autoPurchaseCount = autoPurchaseLottoMoney.getCanPurchaseTicketCount();
+        List<LottoTicket> lottoTickets = new ArrayList<>(
+                autoGenerateTickets(numberGenerateStrategy, autoPurchaseCount));
+        return new LottoTickets(lottoTickets, 0);
+    }
+
+    public static LottoTickets fromTicketNumbers(List<Set<Integer>> ticketNumbers) {
+        List<LottoTicket> lottoTickets = boxTicket(ticketNumbers);
+        return new LottoTickets(lottoTickets, ticketNumbers.size());
+    }
+
+    private static List<LottoTicket> boxTicket(List<Set<Integer>> selfTicketNumbers) {
+        return selfTicketNumbers.stream()
+                .map(LottoTicket::fromNumberValues)
+                .collect(Collectors.toList());
+    }
+
+    private static List<LottoTicket> autoGenerateTickets(NumberGenerateStrategy numberGenerateStrategy,
+                                                         int autoPurchaseCount) {
+        return IntStream.rangeClosed(1, autoPurchaseCount)
+                .mapToObj(index -> LottoTicket.fromNumberValues(numberGenerateStrategy.generateNumbers()))
+                .collect(Collectors.toList());
+    }
+
+    public LottoTickets concat(LottoTickets other) {
+        List<LottoTicket> concatTickets = new ArrayList<>(lottoTickets);
+        concatTickets.addAll(other.getTickets());
+        return new LottoTickets(concatTickets, selfPurchaseCount + other.getSelfPurchaseCount());
+    }
+
+    public List<LottoTicket> getTickets() {
+        return List.copyOf(lottoTickets);
+    }
```

Comment:
> dm으로 불변에 대한 얘기를 해주셔서 혹시나 제가 불변이어야 하는 걸 불변으로 만들지 않았나 싶어 쭉 봤는데요, 혹시나 제가 복사에 실수를 하고 있나 찾아봤는데 처음엔 제가 아는 지식으로는 찾을 수 없었습니다. 

제가 불변에 대해 말씀드릴 당시 LottoGame이 불변객체가 아니었고, 충분히 개선할 수 있을 것 같아 말씀드렸었습니다. 😃  https://github.com/woowacourse/java-lotto/pull/453#discussion_r818258886 

> 또링이 보내준 링크의 글을 보고 Collections.unmodifiableXX는 새로 받은 주소가 기존 컬렉션과 상관 관계가 있어서 새로 받은 곳에서 수정은 막아주지만, 원본 컬렉션을 변경하면 unmodifiableXX로 생성된 컬렉션은 영향을 받는다는걸 알게 되었습니다. 저는 방어적 복사를 하고 있다고 생각했고 컬렉션이 다루는 클래스도 불변으로 설계된 클래스여서 불변이라고 생각했는데, 원본 컬렉션과 관계가 끊어지지를 않았군요. 그래서 기존 Collections.unmodifiableXX을 사용하던 것들을 copyOf로 변경했어요. 또링 덕분에 하나 더 배울 수 있었습니다.

많은 내용을 학습 하셨네요! ㅎㅎ💯 다만 여기서는 한 가지 더 고민해보면 좋을 것 같아요. 
```suggestion
    public List<LottoTicket> getTickets() {
        return Collections.unmodifiableList(lottoTickets);
    }
```
여기서 정말 깊은복사가 필요할까요? 깊은 복사는 어떤 경우에 해야 하는 걸까요?

Reply:
위 경우엔 이미 LottoTicket이 불변인 객체이기 때문에 `unmodifiableList()`로 리스트의 변경만 막아도 충분히 불변성을 가질 것 같아요! 다시 원래대로 `unmodifiableList()`를 사용했습니다.